core/ROB.cpp

```diff
@@ -248,4 +243,44 @@ namespace olympia
         }
     }
 
+    // sys instr doesn't have a pipe so we handle special stuff here
+    void ROB::retireSysInst(InstPtr &ex_inst)
+    {
+        auto reg_file = ex_inst->getRenameData().getDestination().rf;
+        if (reg_file == core_types::RegFile::RF_INVALID)
+        {   // this is the case if dst = x0
+            DLOG("retiring SYS instr ");
+        } else   // this is needed or else destination register is 
+        {        // forever reserved and not ready
+            const auto & dest_bits = ex_inst->getDestRegisterBitMask(reg_file);
+            scoreboard_views_[reg_file]->setReady(dest_bits);
+            DLOG("retiring SYS inst dest reg bits: " <<  sparta::printBitSet(dest_bits));
+        }
+
+        FlushManager::FlushingCriteria criteria(FlushManager::FlushCause::POST_SYNC, ex_inst);
+                    out_retire_flush_.send(criteria);
+                    expect_flush_ = true;
+                    ++num_flushes_;
+                    ILOG("Instigating flush due to SYS instruction... " << *ex_inst);
+
+
+    }
+
+    // for SYS instr which doesn't have an exe pipe
+    void ROB::setup_scoreboard_view()
+    {
+        const char QNAME[] =  "iq0";
```

Comment:
I tried it and the following line of code

 olympia::coreutils::getPipeTopology(getContainer()->getRoot(), "exe_pipe_rename");

throws an exception, somewhere inside the call getPipTopology, off the top of your head any idea why?

Reply:
The issue is that `getContainer()->getRoot()` returns a `RootTreeNode`, while `getPipeTopology()` requires a `TreeNode`. A solution could be since we're in the ROB to just get the parent, which is `cpu.core0` (from `cpu.core0.ROB` and then call the search for the `issue_queue_rename` parameter from there.

So change `getContainer()->getRoot()` to `getContainer()->getParent()`

@klingaard is there a more elegant way to approach this? My initial thoughts was to go from root to ensure we can find the parameter, but due to the `TreeNode` type requirement on `getPipeTopology()`, we would have to step down to core, which we could hard code to `getRoot()->getChild("cpu.core0")`. This could have issues in the future if Olympia is extended to multicore.

Reply:
ok that works thanks Aaron